artifacts_test.py

```diff
@@ -37,7 +37,8 @@ def setUp(self) -> None:
         self.housekeeping.connect()
 
     def tearDown(self) -> None:
-        self.housekeeping.close()
+        if getattr(self, "housekeeping"):
```

Comment:
```
Help on built-in function getattr in module __builtin__:

getattr(...)
    getattr(object, name[, default]) -> value
    
    Get a named attribute from an object; getattr(x, 'y') is equivalent to x.y.
    When a default argument is given, it is returned when the attribute doesn't
    exist; without it, an exception is raised in that case.
```

It should be `getattr(self, "housekeeping", None)`

Reply:
fixed